src/BlobField/blobs.js

```diff
@@ -0,0 +1,210 @@
+			currBall.path.onMouseLeave = this.pathOnMouseLeave.bind(this);
+			currBall.path.onClick = this.pathOnClick.bind(this);
+			this.balls.push(currBall);
+		}
+	}
+
+	onFrame() {
+		for (let i = 0; i < this.balls.length - 1; i++)
+			for (let j = i + 1; j < this.balls.length; j++)
+				this.balls[i].react(this.balls[j]);
+
+		for (let i = 1; i < this.balls.length; i++) {
+			this.balls[i].iterate();
+			this.balls[i].updateColor();
+		}
+
+		this.balls[0].updateShape();
+		this.balls[0].updateColor();
+
+		// Mouse Easing
+		let easeFactor = this.defaultEasing;
+		if (this.isSlowSim) {
+			easeFactor = this.slowSimEasing;
+		}
+		const dx = this.mouseTargetX - this.mouseCurrX;
+		this.mouseCurrX += dx * easeFactor;
+		const dy = this.mouseTargetY - this.mouseCurrY;
+		this.mouseCurrY += dy * easeFactor;
+
+		// Move the hidden/mouse blob
+		if (this.collapsed) {
+			this.balls[0].point = new paper.Point(
+				-this.balls[0].radius,
+				-this.balls[0].radius
+			);
+		} else {
+			this.balls[0].point = new paper.Point(this.mouseCurrX, this.mouseCurrY);
+		}
+	}
+
+	onResize() {
+		for (let i = 0; i < this.balls.length; i++) {
+			this.balls[i].radius = this.calcRadius(i);
+			let tempIsVert = paper.view.size.width > paper.view.size.height;
+			this.balls[i].isVertical = tempIsVert;
+		}
+		this.balls[0].radius = this.calcRadius(0) * this.mouseRadiusMultiplier;
+	}
+
+	onMouseMove(x,y) {
+		this.mouseTargetX = x;
+		this.mouseTargetY = y;
+	}
+
+	pathOnMouseEnter(event) {
+		this.isSlowSim = true;
+
+		const idx = event.target.idx;
+		this.balls[idx].mouseEnterPt = event.point;
+
+		for (let i = 1; i < this.balls.length; i++) {
+			if (this.balls[i].path !== event.target) {
+				this.balls[i].path.tween({
+						opacity: this.balls[i].path.opacity,
+						shadowColor: this.balls[i].path.shadowColor,
+					}, {
+						opacity: this.hoverFadedOpacity,
+						shadowColor: this.balls[i].shadowInactiveColor,
+					},
+					this.animationDuration
+				);
+			} else {
+				this.balls[i].path.tween({
+						opacity: this.balls[i].path.opacity,
+						// shadowColor: this.balls[i].path.shadowColor,
+					}, {
+						opacity: this.hoverActiveOpacity,
+						// shadowColor: this.balls[i].shadowColor,
+					},
+					this.animationDuration
+				);
+			}
+    }
+    this.onArtistHovered && this.onArtistHovered(event.target.artist);
+	}
+
+	pathOnMouseLeave(event) {
+		this.isSlowSim = false;
+
+		const idx = event.target.idx;
+		this.balls[idx].mouseLeavePt = event.point;
+
+		// Add force on mouse leave
+		// this.repulseBall(idx);
+
+		for (let i = 1; i < this.balls.length; i++) {
+			this.balls[i].path.tween({
+					opacity: this.balls[i].path.opacity,
+					shadowColor: this.balls[i].path.shadowColor,
+				}, {
+					opacity: this.opacity,
+					shadowColor: this.balls[i].shadowInactiveColor
+				},
+				this.animationDuration
+			);
+		}
+
+    this.onArtistHovered && this.onArtistHovered();
```

Comment:
for some reason this gets called on every render, even if the cursor is inside the path

Reply:
I can't seem to reproduce this. Is it resolved?

Reply:
still happening on my end. You can't really see it in the UI b/c it immediately gets overwritten by the call in `#pathOnMouseEnter`. But if you open up the browser inspector you'll see it in the logs

Reply:
it is possible it's a weird React thing. I haven't made any effort to debug.

Reply:
I just pushed a commit that does some brute force console logging. Still behaves normally on my end. Fired once on enter and once of leaving.

Reply:
Interesting. I will take another look tomorrow when I'm less brain-fried. thanks!

Reply:
No worries. In my commit I changed the signature of your onArtistX functions to take in the event instead of the artist. This was just for testing purposes but I think it might be useful if we want to do something with the path itself, beyond the artist object.